docs/website/blog/2024-01-11-dlt-dbt-semantic-layer.md

```diff
@@ -0,0 +1,135 @@
+</aside>
+
+## The Chinese Whisper of Data
+
+In the context of constructing a **modern data stack** through the development of various modular components for a data pipeline, our attention turns to the centralization of metrics and their definitions.
+
+For the purposes of this demo, we’ll be looking specifically at how dlt and dbt come together to solve the problem of the data flow from data engineer → analytics engineer → data analyst → business user. That’s quite a journey. And just like any game of *Chinese whisper*, things certainly do get lost in translation.
+
+<div style={{ paddingRight: '10%', paddingLeft: '10%', paddingBottom: '1%' }}>
+
+![cover](/img/blog-dbt_sem-data-people.png)
+<div>Taken from the real or fictitious book called '5th grade data engineering, 1998'.</div>
+
+</div>
+
+To solve for this problem, both these tools come together and seamlessly integrate to create everything from data sources to uniform metric definitions, that can be handled centrally, and hence are a big aid to the data democracy practices of your company!
+
+Here’s how a pipeline could look:
+1. Extract and load with dlt: Dlt will automate data cleaning and normalization leaving you with clean data you can just use.
+2. Create SQL models that simplify sources, if needed. This can include renaming and/or eliminating columns, identifying and setting down key constraints, fixing data types, etc.
+3. Create and manage central metric definitions with the semantic layer.
+
+## 1. Extract, Structure, & Load with dlt
+
+The data being used is of a questionnaire, which includes questions, the options of those questions, respondents and responses. This data is contained within a nested json object, that we’ll pass as a raw source to dlt to structure, normalize and dump into a BigQuery destination.
+
+```python
+# initializing the dlt pipeline with your data warehouse destination
+pipeline = dlt.pipeline(
+        pipeline_name="survey_pipeline",
+        destination="bigquery",
+        dataset_name="questionnaire")
+
+# running the pipeline (into a structured model)
+# the dataset variable contains unstructured data
+pipeline.run(dataset, table_name='survey')
+```
+
+The extract, and load steps of an ETL pipeline have been taken care of with these steps. Here’s what the final structure looks like in BigQuery:
+
+![bigquery tables](/img/blog-dbt_sem-data-bqtables.png)
+
+`questionnaire` is a well structured dataset with a base table, and child tables. The `survey__questions` and `survey_questions__options` are normalized tables with, the individual questions and options of those questions, respectively, connected by a foreign key. The same structure is followed with the `..__respondents` tables, with `survey__respondents__responses` as our fact table.
```

Comment:
do you have any github repo with this dataset source? it would be cool to link it here

Reply:
this was a gpt generated dataset - since adrain suggested we don't have to worry about data sources in this article

Reply:
got it :ok_hand: